src/main/kotlin/org/rust/ide/inspections/RsWrongTypeParametersNumberInspection.kt

```diff
@@ -16,18 +17,74 @@ class RsWrongTypeParametersNumberInspection : RsLocalInspectionTool() {
                 // Don't apply generic declaration checks to Fn-traits and `Self`
                 if (type.path?.valueParameterList != null) return
                 if (type.path?.cself != null) return
+                checkMethod(holder, type)
+            }
+
+            override fun visitCallExpr(o: RsCallExpr) {
+                checkMethod(holder, o)
+            }
```

Comment:
Could be a one-liner:
`override fun visitCallExpr(o: RsCallExpr) = checkMethod(holder, o)`